test/infrastructure/docker/examples/simple-cluster.yaml

```diff
@@ -1,11 +1,11 @@
 # Creates a cluster with one control-plane node and one worker node
-apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
```

Comment:
Trying to apply this config in my tilt setup I am seeing this error in the capd controller
```
[manager] E0329 14:51:58.051663      10 controller.go:302] controller-runtime/manager/controller/dockercluster "msg"="Reconciler error" "error"="DockerCluster.infrastructure.cluster.x-k8s.io \"my-cluster\" is invalid: spec.controlPlaneEndpoint.host: Required value" "name"="my-cluster" "namespace"="default" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="DockerCluster" 
```

Reply:
`spec.controlPlaneEndpoint.host` seems like a field that will be populated by KCP controller, not sure why that's in the `DockerCluster` spec though.

Reply:
@fabriziopandini @CecileRobertMichon any ideas on this? Looks like possibly something needs to change in the example files?

Reply:
not entirely sure, looking at https://github.com/kubernetes-sigs/cluster-api/blob/master/test/e2e/data/infrastructure-docker/v1alpha4/bases/cluster-with-kcp.yaml (which is used by https://github.com/kubernetes-sigs/cluster-api/blob/master/test/e2e/data/infrastructure-docker/v1alpha4/cluster-template/kustomization.yaml), it doesn't look like it's needed in `DockerCluster` spec

Reply:
What's even more confusing is that
https://github.com/kubernetes-sigs/cluster-api/blob/master/test/infrastructure/docker/controllers/dockercluster_controller.go#L150-L153

This is set in the DockerCluster controller.

Are we missing a patch here?

Reply:
Actually, we defer the patch in the top-level Reconcile functions.

Reply:
I think I know what's happening.
We are missing at least one error condition check. I think this PR is good to merge.

Reply:
Basically https://github.com/kubernetes-sigs/cluster-api/blob/master/test/infrastructure/docker/controllers/dockercluster_controller.go#L75 is returning an LB with no hostname and no error.

Reply:
I'll do more digging.

Reply:
Thanks Ashish. Just let me know if you find any need to change something here. Happy to update to fix anything if it's not significant enough to warrant its own Issue/PR.

Reply:
The root cause here is that I had a stopped ha proxy container that was preventing a new one from being created.
During creation of the loadbalancer helper we check if the ha-proxy container exists which is where we pick up stopped containers https://github.com/kubernetes-sigs/cluster-api/blob/master/test/infrastructure/docker/docker/util.go#L108

cc @fabriziopandini going to create an issue from this comment.